test/libsolidity/semanticTests/cleanup/cleanup_bytes_types_shortening_OldCodeGen.sol

```diff
@@ -0,0 +1,13 @@
+contract C {
+    function f() public pure returns (bytes32 r) {
+        bytes4 x = 0xffffffff;
+        bytes2 y = bytes2(x);
+        assembly {
+            r := y
+        }
```

Comment:
I only meant to remove the
```
        // At this point, r and y both store four bytes, but
        // y is properly cleaned before the equality check
        require(y == bytes2(0xffff));
```
part in the new code gen version. In the test for old code gen we should keep that.